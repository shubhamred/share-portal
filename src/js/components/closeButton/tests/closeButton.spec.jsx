import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import CloseButton from '../closeButton';

configure({ adapter: new Adapter() });

describe('components/Close Button', () => {
  it('Renders default options', () => {
    const component = shallow(
      <CloseButton />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('handles click', () => {
    const props = {
      onCloseButtonClick: jest.fn()
    };
    const component = shallow(
      <CloseButton onCloseButtonClick={props.onCloseButtonClick} />
    );
    component.simulate('click');
    expect(props.onCloseButtonClick).toHaveBeenCalledTimes(1);
  });
});
