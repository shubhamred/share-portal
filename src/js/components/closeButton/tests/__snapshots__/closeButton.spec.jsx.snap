// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Close Button Renders default options 1`] = `
<div
  className="closeButton"
  onClick={[Function]}
  role="presentation"
>
  <Memo(ForwardRef(CancelIcon))
    style={
      Object {
        "fill": "#3b3b3b",
        "fontSize": "20",
      }
    }
  />
  <span
    style={
      Object {
        "fontWeight": 100,
      }
    }
  >
    Remove
  </span>
</div>
`;
